Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,344 @@
+            // Check if the current ray hits a magnetic surface
+            DidHitSurface = MixedRealityRaycaster.RaycastSimplePhysicsStep(CurrentRay, MaxRaycastDistance, MagneticSurfaces, false, out CurrentHit);  
+        }
+
+        protected virtual void SetPosition()
+        {
+            // Change the position of the gameObject if there was a hit, if not then place the object at the default distance
+            // relative to the TrackedTargetType origin position
+            
+            if (DidHitSurface)
+            {
+                // Take the current hit point and add an offset relative to the surface to avoid half of the object in the surface
+                GoalPosition = CurrentHit.point;  
+                AddOffset(CurrentHit.normal * SurfaceNormalOffset);
+
+                #if UNITY_EDITOR
+                if(DebugEnabled)
+                {
+                    // Draw the normal of the raycast hit for debugging 
+                    Debug.DrawRay(CurrentHit.point, CurrentHit.normal * 0.5f, Color.yellow);
+                }
+                #endif
+            }
+            else
+            {
+                GoalPosition = SolverHandler.TransformTarget.position + (SolverHandler.TransformTarget.forward * DefaultPlacementDistance);
+            }
+        }
+
+        protected virtual void SetRotation()
+        {
+            Vector3 direction = CurrentRay.Direction;
+            Vector3 surfaceNormal = CurrentHit.normal;
+            
+            if (KeepOrientationVertical)
+            {
+                direction.y = 0;
+                surfaceNormal.y = 0;
+            }
+
+            // If the object is on a surface then change the rotation according to the normal of the hit point
+            if (DidHitSurface && rotateAccordingToSurface)
+            {
+                GoalRotation = Quaternion.LookRotation(-surfaceNormal, Vector3.up);
+            }
+            else 
+            {
+                // If (DidHit and !rotateAccordingToSurface) or if there is no raycast hit, 
+                GoalRotation = Quaternion.LookRotation(direction, Vector3.up);
+            }
+        }
+
+        #region IMixedRealityPointerHandler
+
+        /// <inheritdoc/>
+        public void OnPointerDown(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerDragged(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerUp(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerClicked(MixedRealityPointerEventData eventData)
+        {
+            // When a click is called in the same second then it is a mistake and no action needs to be taken
+            if ((Time.time - TimeSinceClick) < 1.0f)
+            {
+                return;
+            }
+
+            if (!IsBeingPlaced)
+            {
+                StartPlacement();
+                OnPlacingStarted?.Invoke();
```

Comment:
As above, recommend just invoking these events directly in Start/Stopplacement to avoid needing to remember to invoke the events every time.

Reply:
Placing events have been moved into StartPlacement and StopPlacement.